src/Microsoft.ServiceFabric.Services/Communication/Client/ServicePartitionClient.cs

```diff
@@ -124,6 +124,9 @@ public bool TryGetLastResolvedServicePartition(out ResolvedServicePartition reso
         /// Invokes the given Function, retrying for exceptions thrown other than the exceptions in the doNotRetryExceptionTypes.
         /// For exceptions that are not in doNotRetryExceptionTypes, CommunicationClientFactory's ReportOperationExceptionAsync() method 
         /// controls if the exception should be retried or not.
+        /// If you are invoking this method in Asp.Net / UI thread, these are recommendations to avoid deadlock:
+        ///  1 if your calling Api is Async , configure await to not to resume in orignal context. by setting ConfigureAwait as false. 

```

Comment:
use <see cref="system.threading.tasks.task.configureawait" /> to not resume in the original context by setting it to false.